io/net/irqbalance.py

```diff
@@ -0,0 +1,437 @@
+            return False
+        return process_pid
+
+    def compare_range_strings(self, range_str1, range_str2):
+        '''
+        Function to compare and match cpu range of string format.
+        :rtype : boolean value.
+        '''
+        range_pattern = r'(\d+)-(\d+)'
+        match1 = re.match(range_pattern, range_str1)
+        match2 = re.match(range_pattern, range_str2)
+
+        if match1 and match2:
+            start1, end1 = int(match1.group(1)), int(match1.group(2))
+            start2, end2 = int(match2.group(1)), int(match2.group(2))
+
+            return start1 == start2 and end1 == end2
+        return False
+
+    def cpu_range_validation(self):
+        '''
+        Funtion to validate the assinged CPU's by script.
+        '''
+        self.irq_affinity = '-'.join([str(self.cpu_range[0]),
+                                      str(self.cpu_range[-1])])
+        cmd = f'/proc/irq/{self.irq_number}/smp_affinity_list'
+        self.system_affinity = genio.read_file(cmd)
+        time.sleep(25)
+        if not self.compare_range_strings(self.irq_affinity,
+                                          self.system_affinity):
+            self.fail(f'The smp_affinity_list does not matches to affinity'
+                      ' that was set by script, Please check the logs')
+
+    def grep_module_interrupts(self):
+        '''
+        Funtion to filter all interrupts along associated CPU's of device.
+        '''
+        cmd = f'head -n 1 /proc/interrupts &&' \
+              f' grep {self.interface_type} /proc/interrupts'
+
+        process.run(cmd, shell=True, ignore_status=True
+                    ).stdout.decode().strip()
+
+    def taskset_cpu_validation(self):
+        '''
+        Function to validate the CPU number set by "taskset" command.
+        Returns : <int> value set by script.
+        '''
+        cmd = "ps -o psr -p %s | awk 'NR>1 {print $1}'" % (
+            self.get_ping_process_pid()
+        )
+        cpu_by_script = process.system_output(cmd, shell=True,
+                                              ignore_status=True,
+                                              sudo=True).decode("utf-8")
+        return int(cpu_by_script)
+
+    def dd_process_pid(self):
+        '''
+        Function to extract PID for running "dd" Process.
+        Returns: PID value.
+        Rtype: int
+        '''
+        cmd = f"ps -C dd | awk 'NR > 1 {{print $1}}'"
+        dd_process_pid = process.system_output(cmd, shell=True,
+                                               ignore_status=True, sudo=True
+                                               ).decode("utf-8")
+        if not dd_process_pid:
+            self.fail(f"No more dd run process PID avaialable")
+        return dd_process_pid
+
+    def dd_run(self):
+        '''
+        Runs the dd command on given Disk and returns True or False
+        '''
+        cmd = f"dd if=/dev/urandom of={self.disk} bs=1M count=$((180*60))"
```

Comment:
what is the logic behind  180*60?

Reply:
hey the dd command must be run for 3 minutes to get the pid  so the command should be alive for 3 minutes and that requires 180*60  count

Reply:
```
dd if=/dev/urandom of=/dev/mapper/mpathj bs=1M count=$((180*60))
10800+0 records in
10800+0 records out
11324620800 bytes (11 GB, 11 GiB) copied, 41.0607 s, 276 MB/s
```

Reply:
@abdhaleegit @Naresh-ibm 
Regarding the timelines, have increased the dd_run to run at-least for 2:30 Min,
sometimes the timings depends on type of device which we are using, the below examples are of NVME based devices.

However, after setting the "CPU" by "taskset" the script kills the dd_run to validate the cpu number that set by script.
in any case dd_run is required only to run for less than 1 min.

we still reduce the count to 1:30 min if required.

---------------------- DD run with dif count values ----------------------------------------------
[root@ltcden7-lp14 net]# time dd if=/dev/urandom of=/dev/nvme0n1 bs=1M count=$((600*60))
36000+0 records in
36000+0 records out
37748736000 bytes (38 GB, 35 GiB) copied, 144.343 s, 262 MB/s

real	2m24.344s
user	0m0.000s
sys	2m23.591s

[root@ltcden7-lp14 net]# time dd if=/dev/urandom of=/dev/nvme0n1 bs=1M count=$((400*60))
24000+0 records in
24000+0 records out
25165824000 bytes (25 GB, 23 GiB) copied, 95.716 s, 263 MB/s

real	1m35.718s
user	0m0.010s
sys	1m34.495s

--------------------- Ends here ----------------------------------------------------------------------

so the logic behind "600*60" to keep run the dd_run for some time until "taskset" change the CPU by its PID.

Reply:
@Naresh-ibm any comments ?

Reply:
@FarooqAbdulla01 dd process is limited to the size of disks... like for a 10G disk it will end with in 5 secs

Reply:
try running same code for a  smalle size non nvme disks

Reply:
> @Naresh-ibm any comments ?

NO